packages/expo-updates/src/UpdatesProvider.tsx

```diff
@@ -0,0 +1,237 @@
+ * Provided to application code from the [`useUpdates`](#useupdatesprovidereventhandler) hook.
+ */
+const runUpdate = () => {
+  throw new Error(
+    'This error occurs when an application directly imports this method from the module. To use this method, it must be provided by the useUpdates() hook.'
+  );
+};
+
+/**
+ * Calls `Updates.checkForUpdateAsync()` and uses the passed in setter
+ * to refresh the [`UpdatesInfo`](#updatesinfo).
+ * Provided to application code from the [`useUpdates`](#useupdatesprovidereventhandler) hook.
+ */
+const checkForUpdate = () => {
+  throw new Error(
+    'This error occurs when an application directly imports this method from the module. To use this method, it must be provided by the useUpdates() hook.'
+  );
+};
+
+/**
+ * Calls `Updates.readLogEntriesAsync()` and sets the `logEntries` property in the [`UpdatesInfo`](#updatesinfo) structure to the results.
+ * Provided to application code from the [`useUpdates`](#useupdatesprovidereventhandler) hook.
+ *
+ * @param maxAge Sets the max age of retrieved log entries in milliseconds. Default to 3600000 ms (1 hour).
+ */
+const readLogEntries = (maxAge?: number) => {
+  throw new Error(
+    'This error occurs when an application directly imports this method from the module. To use this method, it must be provided by the useUpdates() hook.'
+  );
+};
+
+/////// Provider and hook ///////////
+
+/**
+ * Provides the Updates React context. Includes an [`UpdateEvent`](#updateevent) listener
+ * that will set the context automatically, if automatic updates are enabled and a new
+ * update is available. This is required if application code uses the [`useUpdates`](#useupdatesprovidereventhandler) hook.
+ * @param props Context will be provided to `props.children`
+ * @returns the provider.
+ * @example
+ * ```jsx App.tsx
+ * import * as Updates from 'expo-updates';
+ *
+ * const { UpdatesProvider } = Updates.Provider;
+ *
+ * import UpdatesDemo from './src/UpdatesDemo';
+ *
+ * export default function App() {
+ *   return (
+ *     <UpdatesProvider>
+ *       <UpdatesDemo />
+ *     </UpdatesProvider>
+ *   );
+ * }
+ * ```
+ */
+const UpdatesProvider = (props: { children: any }) => {
+  const [updatesInfo, setUpdatesInfo] = useState({
+    currentlyRunning,
+  });
+  // Set up listener for events from automatic update requests
+  // that happen on startup, and use events to refresh the updates info
+  // context
+  useUpdateEvents((event) => {
+    setUpdatesInfo(updatesInfoFromEvent(event));
+  });
+  return (
+    <UpdatesContext.Provider value={{ updatesInfo, setUpdatesInfo }}>
+      {props.children}
+    </UpdatesContext.Provider>
+  );
+};
+
+/**
+ * Hook that obtains the Updates info structure and functions.
+ * Requires that application code be inside an [`UpdatesProvider`](#updatesproviderprops).
+ * @param providerEventHandler Optional handler. If present, the handler will be called on
```

Comment:
I'm curious what use cases you envision for this event handler? I see it at https://github.com/douglowder/CriticalUpdatesTest/blob/main/src/UpdatesDemo.tsx#L23 but am not entirely clear what that is doing. 

In general, having to mix event bus architecture with hooks and react seems to me to be a bit confusing. Maybe there's a way we can express these elsewhere (args to download or something)?

Reply:
I put this in for the use case where we eventually might want to show a download progress bar for a large update. Maybe we can separate that out and use the native event emitter pattern instead.

Reply:
Thought about this some more... I do believe that this is the right pattern, but it should be termed a "callback" and not an "event handler". I'll rename the types and the handler and provide better descriptions.

Reply:
> I put this in for the use case where we eventually might want to show a download progress bar for a large update. Maybe we can separate that out and use the native event emitter pattern instead.

I would assume this would be a state variable.